tuf/client/client_test.go

```diff
@@ -330,6 +330,49 @@ func TestDownloadTargetsHappy(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+func TestDownloadTargetsDeepHappy(t *testing.T) {
```

Comment:
Wondering if we can make this a deeper and wider tree?  (maybe targets, targets/level1, targets/level2, targets/level1/level3, targets/level1/level4, to test (1) that it does all the children delegations for each role that it loops through, and (2) that it can do more than 1 level?)